src/test/scala/com/databricks/spark/xml/XmlSuite.scala

```diff
@@ -1009,4 +1012,97 @@ final class XmlSuite extends FunSuite with BeforeAndAfterAll {
       <integer_value int="Ten">Ten</integer_value>.toString))
   }
 
+
+  test("from_xml roundtrip happy path") {
+
+    val xmlData =
+      s"""
+         |   <parent><pid>14ft3</pid>
+         |      <name>dave guy</name>
+         |   </parent>
+       """.stripMargin
+
+    val xmlSchema: StructType = new StructType().add("pid", StringType).add("name", StringType)
+    val rowSchema: StructType = StructType(
+      Seq(StructField("number", IntegerType, true), StructField("payload", StringType, true)))
+    val expectedSchema: StructType = rowSchema.add("decoded", xmlSchema)
+
+    val df: DataFrame = spark.createDataFrame(
+      spark.sparkContext.parallelize(List(Row(8, xmlData))), rowSchema)
+
+    val result: DataFrame = df.withColumn("decoded",
+      from_xml(df.col("payload"), xmlSchema))
+
+    assert(expectedSchema == result.schema)
+    assert(result.where(col("decoded").isNotNull).count() > 0)
+  }
+
+  /*  this unit test shows that there is no current support for changing the root tag
+     when using this function. Support may be added later. In which case, this test should pass */
+  test("from_xml roundtrip with rowTag or rootTag fails") {
+
+    val xmlData =
+      s"""
+         |<grandparent>
+         |   <parent><pid>14ft3</pid>
+         |      <name>dave guy</name>
+         |   </parent>
+         |</grandparent>
+       """.stripMargin
+
+    val xmlSchema: StructType = new StructType().add("pid", StringType).add("name", StringType)
+    val rowSchema: StructType = StructType(
+      Seq(StructField("number", IntegerType, true), StructField("payload", StringType, true)))
+    val expectedSchema: StructType = rowSchema.add("decoded", xmlSchema)
+
+    val df: DataFrame = spark.createDataFrame(
+      spark.sparkContext.parallelize(List(Row(8, xmlData))), rowSchema)
+
+    val result: DataFrame = df.withColumn("decoded",
+      from_xml(df.col("payload"), xmlSchema, Map("rowTag" -> "parent", "rootTag" -> "parent")))
+
+    assert(expectedSchema == result.schema)
+    assert(result.where(col("decoded").isNotNull).count() > 0)
+
+    assert(0 == result
+      .where(col("decoded.pid").isNotNull or col("decoded.name").isNotNull).count())
+  }
+
+
+  test("from_xml array support") {
```

Comment:
Let's don't add array support for now :D. JSON in Spark added the support for an array a long time ago .. and it's there IMHO for legacy reason. To be honest, I doubt it was a great idea yet.